opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -108,9 +108,10 @@ def test_batch_span_processor(self):
         for name in span_names:
             _create_start_and_end_span(name, span_processor)
 
-        span_processor.shutdown()
+        span_processor.force_flush()
```

Comment:
Just something slightly suspicious here. The test cases pass even without the changes for the test cases included in this PR. It seems like before this PR we were testing that `span_names` and `span_names_list` were equal after calling `span_processor.shutdown()`. Now we test the same but after `span_processor.force_flush()`. Nevertheless, the former case would also pass, so this could mean that we are no longer testing something worthy of testing, that `span_names` and `span_names_list` are equal after calling `span_processor.shutdown()`.

Reply:
What we should definitely test, and is missing now, is that after `force_flush`, new(ly ended) spans are still exported properly.
Also, we should have some multi-threaded test cases with the batch span processor (e.g. calling force_flush while the batch span processor is currently exporting spans itself, ...).

Reply:
@ocelotl & @Oberon00 thanks for catching this up, I handled it on https://github.com/open-telemetry/opentelemetry-python/pull/389/commits/4cf3c3c6761b7ca8dbb7f0a0ce91243f1c18bf49.
For the multi-threading test I'll open an issue.

Reply:
looks like the commit keeps them as the same test. I'd argue that every unit of behavior should be tested separately, as it helps someone running them understand exactly where the regressions are, rather than hunting through test code that's longer than it needs to be.

Reply:
I separated the tests to have `test_shutdown()` and `test_force_flush()`